### PR TITLE
copr: port some tests from TiDB and refine some comments

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Problem Summary:

In this [PR](https://github.com/pingcap/tidb/pull/17418), we fix the behavior of `LIKE` and `REGEXP` in TiDB. It seems that the implementation in TiKV is right, so I port the test from TiDB to TiKV.
Besides, refine some outdated comments. 
### Check List 

Tests 
- No code

Side effects

### Release note 
- No Release note

### CODE REVIEW

**Comment by wjhuang2016** [comment]
@breeswish PTAL

File: components/tidb_query_shared_expr/src/like.rs [hunk 1]
```diff
@@ -6,9 +6,6 @@
 //!
 //! 1. It is not effective. Consider target = 'aaaaaaaaaaaaaaa' and pattern = 'a%a%a%a%a%a%b'.
 //!    See https://research.swtch.com/glob
```
Comment (breeswish) [inline components/tidb_query_shared_expr/src/like.rs:8]: This comment is outdated as well. The implementation will resist this case.

**Comment by breeswish** [comment]
@TennyZhuang PTAL

**Comment by breeswish** [comment]
Please also run `make dev` to fix the format.

**Pull request merged**